lib/plug/cowboy/handler.ex

```diff
@@ -15,12 +18,26 @@ defmodule Plug.Cowboy.Handler do
       {:ok, req, {plug, opts}}
     catch
       kind, reason ->
-        exit_on_error(kind, reason, System.stacktrace(), {plug, :call, [conn, opts]})
+        stacktrace = System.stacktrace()
+        metadata = %{kind: kind, error: reason, stacktrace: stacktrace, conn: conn}
```

Comment:
I am not sure I would include the `conn` here if we are not including it on start. Thoughts? Probably better to be safe than sorry and not include it. We can always add it later.

Reply:
Would the user be able to access the current state of the `conn` in another place? Is there anything of debug value in there you can think of or is the stacktrace enough? I don't think it's all that useful at the `start` or even the `stop` in this case, but I could be convinced otherwise.

As an exercise - is there an argument to be made that it's not harmful to include more information since it's essentially "free"?

I don't have a strong position either way on this.

Reply:
The issue is that this `conn` has no "state". This "conn" is the same as the connection given on "start" since failures are non-local returns.  I would consider the purpose of these hooks to be exclusive for measurements collection. I think it is too early to do anything useful with them (unless we emit them later which makes the measurement worse).

Reply:
Yeah I agree it’s probably fine to remove the conn here. For example, we already have a nested handler in Phoenix that has some more context from wrapped errors that include a conn.

Reply:
I'm good to remove it and add it later.

>This "conn" is the same as the connection given on "start"
@josevalim if we want to see something about the conn in the case of failure then we need to have access to it. It's not interesting to us on `start` and we definitely don't want to add it and store it "just in case" you need it in processing a failure. I'm not sure a stacktrace always has enough to work on.

Reply:
@bryannaegele just to make sure we are on the same page, which kind of information do you think would be available in the `conn` when we have a failure?

In any case, if we are going to have it on failure, then I would propose to also have it on start/stop. :)

Reply:
@josevalim I left it out for now but the idea for having these types of things in the meta on exception and stop is that's usually the point where you may conditionally want to record or act on something in the meta. I suppose something could be done with it in a start event but  I haven't seen that case yet.